sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -544,6 +544,14 @@ abstract class JdbcDialect extends Serializable with Logging {
     if (limit > 0 ) s"LIMIT $limit" else ""
   }
 
+  /**
+   * MS SQL Server version of `getLimitClause`.
+   * This is only supported by SQL Server as it uses TOP (N) instead.
```

Comment:
The problem is TOP (N) is just the SQLServer syntax of LIMIT. It's really weird to have a new dialect API for it, as it's different from table sample which is a new feature.